app/controllers/application_controller.rb

```diff
@@ -1,2 +1,49 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  rescue_from ActionController::InvalidAuthenticityToken, with: -> { sign_out current_user }
+
+  rescue_from ActiveRecord::RecordNotFound, with: :page_not_found
+
+  rescue_from Wicked::Wizard::InvalidStepError, with: :page_not_found
+
+  rescue_from Exceptions::NotAuthorized, with: :not_authorized
+
+  helper_method :current_order
+
+  private
+
+  def page_not_found
+    render file: Rails.root.join('public', '404.html'), layout: false, status: :not_found
+  end
+
+  def not_authorized
+    redirect_to(root_path, alert: I18n.t('auth.errors.not_authorized'))
+  end
+
+  def current_order
+    @current_order ||= obtain_order
+  end
+
+  def obtain_order
+    order = Order.find_or_create_by(id: session[:current_order_id])
+    session[:current_order_id] = order.id
+
+    order.decorate
+  end
+
+  def expose_address_forms(result)
```

Comment:
should it be in ApplicationController?

Reply:
Where it can be?

Reply:
`expose_address_forms` method is specific for Address, not the whole application
move it to the Controller where it's used

Reply:
It using in two controllers:

`AddressesController`
`CheckoutStepsController`